auto_circuit/utils/graph_utils.py

```diff
@@ -395,6 +395,28 @@ def mask_fn_mode(model: PatchableModel, mask_fn: MaskFn, dropout_p: float = 0.0)
             wrapper.dropout_layer.p = 0.0  # type: ignore
 
 
+@contextmanager
+def set_mask_batch_size(model: PatchableModel, batch_size: int | None):
+    """
+    Context manager to set the batch size of the patch masks in the model.
+
+    Args:
+        model: The patchable model to alter.
+        batch_size: The batch size to set the patch masks to. If `None`, the batch size
+            is not modified.
+
+    Warning:
+        This function modifies the state of the model! This is a likely source of bugs.
```

Comment:
This needs another warning. That it breaks other functions of the library while the context is active and should be considered an experimental feature.